### PR TITLE
Fix bugzilla 1223442

### PR DESCRIPTION
**Pull request by jlmarr10824**
Had lifetime problems with the module scope variable so switched to
using function scope.

### CODE REVIEW

File: dxr/plugins/clang/condense.py [hunk 1]
```diff
@@ -17,6 +17,10 @@
 from dxr.indexers import FuncSig, Position, Extent
```
Comment (erikrose) [inline dxr/plugins/clang/condense.py]: I notice that both files have had their permissions changed (see "100644 → 100755") on the "Files changed" tab. I'm sure you didn't mean to do that, so have a look at your setup and see what might be changing permissions behind your back. If Windows is involved, that's one place I'd look.

File: dxr/plugins/clang/condense.py [hunk 2]
```diff
@@ -283,10 +287,36 @@ def lines_from_csv(path):
...
+       start with the specified file path digest.
+
+    All lines are lists of strings.
+
+    :arg folder: The folder in which to look for CSVs
+    :arg csvdict: A dict of sha1 hex digest keys
+    :arg path_digest: A string produced by sha1(file_path).hexdigest()
+
+    """
+    def lines_from_csv(path):
```
Comment (erikrose) [inline dxr/plugins/clang/condense.py]: This looks like a copy-and-paste of the same inner function above; let's factor it out and not repeat ourselves.

**Comment by erikrose** [comment]
Did you happen to try the radix method mentioned in https://bugzilla.mozilla.org/show_bug.cgi?id=1223442#c0? I wonder how it compares for speed with this; it would certainly use less RAM, though it would do more IO in calling listdir() all the time.

**Comment by erikrose** [comment]
I'm having a little swing at refactoring this. It's a good start. More tomorrow!

**Comment by jlmarr10824** [comment]
- I believe I changed the permissions while I was thrashing about trying to figure out why nosetest was not working for me.  Sorry that made it into the submission.  I assume you can rectify that.
- Thanks for refactoring the copy/paste.
- I didn't try the radix approach.  I've had bad experiences with performance of file system globs in perl, and assumed the same would apply in python.

**Comment by erikrose** [comment]
Here's the refactoring I did: https://github.com/mozilla/dxr/compare/master...erikrose:no-globbing?expand=1. I'm benching it now.

**Comment by jlmarr10824** [comment]
Ok, great.   Since my previous implementation was buggy I'm no longer expecting an order-of-magnitude improvement, but still something significant.  Test still running on my end.

**Comment by erikrose** [comment]
And now I can't get Firefox to build: https://bugzilla.mozilla.org/show_bug.cgi?id=1249405.

**Comment by erikrose** [comment]
@pelmers Could you take a look at this?

File: dxr/plugins/clang/condense.py [hunk 3]
```diff
@@ -342,6 +371,15 @@ def listify_keys(d):
     parents = {}
     children = {}
 
+    # create a list of csvfiles then populate dict 'db' with a list of csv's for each filepath digest
+    csvfiles = None
```
Comment (pelmers) [inline dxr/plugins/clang/condense.py]: This line looks unnecessary

**Comment by pelmers** [comment]
I ran Erik's no-globbing branch on a compile of Python and find that it's finishing some 25% faster (3 minute index vs 4 minute index). I'll try to see what the difference is on mozilla-central overnight, but current master completed in 3:17:33 for me.

**Comment by pelmers** [comment]
Tried to run it overnight, but the process hangs at 0% of the "indexing files" stage.

**Comment by erikrose** [comment]
Can you try one more time? There used to be a transient hang with concurrent.futures on Vagrant; it may also exist on Docker.

**Pull request closed without merging**